tests/debugpy/test_evaluate.py

```diff
@@ -677,3 +677,88 @@ def code_to_debug():
 
         session.request_continue()
         assert backchannel.receive() == 1000
+
+
+def test_evaluate_thread_locks(pyfile, target, run):
+    @pyfile
+    def code_to_debug():
+        """
+        The idea here is that a secondary thread does the processing of instructions,
+        so, when all threads are stopped, doing an evaluation for:
+
+        processor.process('xxx')
+
+        would be locked until secondary threads start running.
+        See: https://github.com/microsoft/debugpy/issues/157
+        """
+
+        import threading
+
+        try:
+            from queue import Queue
+        except:
+            from Queue import Queue
+
+        class EchoThread(threading.Thread):
+            def __init__(self, queue):
+                threading.Thread.__init__(self)
+                self._queue = queue
+
+            def run(self):
+                while True:
+                    obj = self._queue.get()
+                    if obj == "finish":
+                        break
+
+                    print("processed", obj.value)
+                    obj.event.set()
+
+        class NotificationObject(object):
+            def __init__(self, value):
+                self.value = value
+                self.event = threading.Event()
+
+        class Processor(object):
+            def __init__(self, queue):
+                self._queue = queue
+
+            def process(self, i):
+                obj = NotificationObject(i)
+                self._queue.put(obj)
+                assert obj.event.wait()
+
+            def finish(self):
+                self._queue.put("finish")
+
+        if __name__ == "__main__":
+            queue = Queue()
+            echo_thread = EchoThread(queue)
+            processor = Processor(queue)
+            echo_thread.start()
+
+            processor.process(1)
+            processor.process(2)  # @bp
+            processor.process(3)
+            processor.finish()
+
+    with debug.Session() as session:
+
+        # During the evaluation we'll actually have continued/stopped events because
+        # we're letting threads run at that time. Let's ignore these in the test.
+        session.ignore_unobserved.extend([timeline.Event("stopped")])
```

Comment:
I wonder how many other tests would be potentially affected by this (e.g. by helper threads spawned by stdlib)? If it's common, we might need to add it to the default exclusion list in session.py.

Reply:
I'm not sure... I think something like this is pretty rare, so, I think adding it per test makes sense.